### PR TITLE
gh-83004: Harden winsound init

### PR DESCRIPTION
**Pull request by erlend-aasland**
* Issue: gh-83004

### CODE REVIEW

**Review by hauntsaninja — COMMENTED** [review]
Nice cleanup! Should we decref module in the error case?

File: PC/winsound.c [hunk 1]
```diff
@@ -254,5 +245,10 @@ PyInit_winsound(void)
     ADD_DEFINE(MB_ICONEXCLAMATION);
     ADD_DEFINE(MB_ICONHAND);
     ADD_DEFINE(MB_ICONQUESTION);
+
```
Comment (erlend-aasland) [inline PC/winsound.c:248]:
```suggestion

#undef ADD_DEFINE

```

Comment (erlend-aasland) [reply PC/winsound.c:248]: Not strictly needed, but I prefer to be explicit with macro lifetime.

Comment (kumaraditya303) [reply PC/winsound.c:248]: SGTM

**Review by kumaraditya303 — APPROVED** [review]

**Comment by erlend-aasland** [comment]
Thanks for the review, Kumar!

**Pull request merged**